TCC/Parser/CommandLineHelper.cs

```diff
@@ -0,0 +1,157 @@
+
+            var parsed = CommandLine.Parser.Default.ParseArguments<CompressCmdOptions, DecompressOptions, BenchmarkOptions>(args);
+            try
+            {
+                returnCode = parsed.MapResult(
+                    (CompressCmdOptions opts) =>
+                    {
+                        mode = Mode.Compress;
+                        option = new CompressOption
+                        {
+                            Algo = CompressionAlgo.Lz4,
+                            DestinationDir = opts.Output,
+                            FailFast = opts.FailFast,
+                            SourceDirOrFile = opts.Source.FirstOrDefault(),
+                            BlockMode = Lib.Blocks.BlockMode.Individual,
+                            Threads = ExtractThreads(opts)
+                        };
+
+                        ExtractPasswordInfo(opts, option, mode);
+
+                        return 0;
+                    },
+                    (DecompressOptions opts) =>
+                    {
+                        mode = Mode.Decompress;
+                        option = new DecompressOption
+                        {
+                            Algo = CompressionAlgo.Lz4,
+                            DestinationDir = opts.Output,
+                            FailFast = opts.FailFast,
+                            SourceDirOrFile = opts.Source.FirstOrDefault(),
+                            Threads = ExtractThreads(opts)
+                        };
+
+                        ExtractPasswordInfo(opts, option, mode);
+
+                        return 0;
+                    },
+                    (BenchmarkOptions opts) =>
+                    {
+                        mode = Mode.Benchmark;
+                        option = new DecompressOption();
+                        return 0;
+                    },
+                    errs => 1);
+            }
+            catch (CommandLineException ae)
+            {
+                Console.Out.WriteLine(ae.Message);
+                returnCode = 1;
+            }
+            catch (Exception e)
+            {
+                Console.Out.WriteLine(e.ToString());
+                returnCode = 1;
+            }
+            return (mode, option, returnCode);
+        }
+
+        private static int ExtractThreads(BaseCmdOptions opts)
+        {
+            if (string.IsNullOrEmpty(opts.Threads))
+            {
+                return 1;
+            }
+            if (string.Equals(opts.Threads, "all", StringComparison.InvariantCultureIgnoreCase))
+            {
+                return Environment.ProcessorCount;
+            }
+            if (int.TryParse(opts.Threads, out var nbThread))
+            {
+                return nbThread;
+            }
+            throw new CommandLineException("Maximum threads need to be either numeric, or \"all\" ");
+        }
+
+        private static void ExtractPasswordInfo(BaseCmdOptions opts, TccOption option, Mode mode)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 21 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)